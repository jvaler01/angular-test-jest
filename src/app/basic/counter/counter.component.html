<h2>Counter {{ counter }}</h2>
<hr>


<button class="btn btn-primary mx-1" (click)="increaseBy(+1)">+1</button>
<button class="btn btn-primary mx-1" (click)="increaseBy(-1)">-1</button>
