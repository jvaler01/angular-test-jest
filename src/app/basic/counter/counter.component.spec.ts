import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CounterComponent } from './counter.component';

describe('CounterComponent', () => {
  let component: CounterComponent;
  let fixture: ComponentFixture<CounterComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ CounterComponent ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(CounterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('check snapshoot', () => {
    expect( compiled ).toMatchSnapshot();
  });

  it('check increaseBy increment by 5', () => {
    component.increaseBy(5);
    expect( component.counter ).toBe(15);
  });

  it('check on click buttons increase 1 and decrease 1', () => {
    const buttons = compiled.querySelectorAll('button');
    buttons[0].click();
    expect( component.counter ).toBe(11);
    buttons[1].click();
    buttons[1].click();
    expect( component.counter ).toBe(9);
    
  });

  it('check increaseBy update counter h2 value', () => {
    component.increaseBy(10);
    fixture.detectChanges();

    const h2 = compiled.querySelector('h2');
    expect( h2?.textContent ).toContain('20');
  });
});
