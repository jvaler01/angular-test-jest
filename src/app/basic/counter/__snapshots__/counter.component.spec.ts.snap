// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterComponent check snapshoot 1`] = `
<div
  id="root1"
>
  <h2>
    Counter 10
  </h2>
  <hr />
  <button
    class="btn btn-primary mx-1"
  >
    +1
  </button>
  <button
    class="btn btn-primary mx-1"
  >
    -1
  </button>
</div>
`;
